controller_manager:
  ros__parameters:
    update_rate: 1000
    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster
    joint_trajectory_controller:
      type: joint_trajectory_controller/JointTrajectoryController
    cartesian_motion_controller:
      type: cartesian_motion_controller/CartesianMotionController
    cartesian_force_controller:
      type: cartesian_force_controller/CartesianForceController
    joint_effort_controller:
      type: effort_controllers/JointGroupEffortController
    gripper_controller:
      type: position_controllers/GripperActionController
    force_torque_sensor_broadcaster:
      type: force_torque_sensor_broadcaster/ForceTorqueSensorBroadcaster

force_torque_sensor_broadcaster:
  ros__parameters:
    frame_id:  "tool0"
    sensor_name: "fts_sensor"

joint_trajectory_controller:
  ros__parameters:
    joints:
      - fork0_link_joint
      - main0_link_joint
      - fork1_link_joint
      - main1_link_joint
      - fork2_link_joint
      - ee_link_joint
      - rbs_gripper_rot_base_joint
    command_interfaces:
      - position
    state_interfaces:
      - position
      - velocity
    state_publish_rate: 100.0
    action_monitor_rate: 20.0
    allow_partial_joints_goal: false

cartesian_motion_controller:
  ros__parameters:
    end_effector_link: "gripper_grasp_point"
    robot_base_link: "base_link"
    joints:
      - fork0_link_joint
      - main0_link_joint
      - fork1_link_joint
      - main1_link_joint
      - fork2_link_joint
      - ee_link_joint
      - rbs_gripper_rot_base_joint
    command_interfaces:
      - position
    solver:
      error_scale: 1.0
      iterations: 10
      publish_state_feedback: True
    pd_gains:
      trans_x: { p: 1.0 }
      trans_y: { p: 1.0 }
      trans_z: { p: 1.0 }
      rot_x: { p: 0.5 }
      rot_y: { p: 0.5 }
      rot_z: { p: 0.5 }


joint_effort_controller:
  ros__parameters:
    joints:
      - fork0_link_joint
      - main0_link_joint
      - fork1_link_joint
      - main1_link_joint
      - fork2_link_joint
      - ee_link_joint
      - rbs_gripper_rot_base_joint
  
gripper_controller:
  ros__parameters:
    action_monitor_rate: 200.0
    joint: rbs_gripper_r_finger_joint
    goal_tolerance: 0.01
    max_effort: 5.0
    allow_stalling: false
    stall_velocity_threshold: 0.001
    stall_timeout: 2.0


cartesian_force_controller:
  ros__parameters:
    end_effector_link: "gripper_grasp_point"
    robot_base_link: "base_link"
    ft_sensor_ref_link: "tool0"
    ft_sensor_node_name: "force_torque_sensor_broadcaster"
    joints:
      - fork0_link_joint
      - main0_link_joint
      - fork1_link_joint
      - main1_link_joint
      - fork2_link_joint
      - ee_link_joint
      - rbs_gripper_rot_base_joint

    # Choose: position or velocity.
    command_interfaces:
      # - position
      - velocity

    solver:
        error_scale: 0.5

    pd_gains:
        trans_x: {p: 0.05}
        trans_y: {p: 0.05}
        trans_z: {p: 0.05}
        rot_x: {p: 1.5}
        rot_y: {p: 1.5}
        rot_z: {p: 1.5}
